parlai/core/torch_ranker_agent.py

```diff
@@ -457,13 +449,16 @@ def eval_step(self, batch):
         # Update metrics
         if label_inds is not None:
             loss = self.criterion(scores, label_inds)
-            self.metrics['loss'] += loss.item()
-            self.metrics['examples'] += batchsize
+            self.record_local_metric('loss', AverageMetric.many(loss))
+            ranks_m = []
+            mrrs_m = []
             for b in range(batchsize):
                 rank = (ranks[b] == label_inds[b]).nonzero()
                 rank = rank.item() if len(rank) == 1 else scores.size(1)
-                self.metrics['rank'] += 1 + rank
-                self.metrics['mrr'] += 1.0 / (1 + rank)
+                ranks_m.append(1 + rank)
+                mrrs_m.append(1.0 / (1 + rank))
```

Comment:
I'd be curious to understand this calculation, but that's beyond the scope of this PR =P